interfaces/builtin/camera.go

```diff
@@ -34,11 +36,65 @@ const cameraConnectedPlugAppArmor = `
 /run/udev/data/c81:[0-9]* r, # video4linux (/dev/video*, etc)
 `
 
-// NewCameraInterface returns a new "camera" interface.
-func NewCameraInterface() interfaces.Interface {
-	return &commonInterface{
-		name: "camera",
-		connectedPlugAppArmor: cameraConnectedPlugAppArmor,
-		reservedForOS:         true,
+var cameraPermanentSlotUdev = []byte(`
+# RealSense UVC cameras (R200, F200, SR300 LR200, ZR300)
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="8086", ATTRS{idProduct}=="0a80", MODE:="0666", GROUP:="plugdev"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="8086", ATTRS{idProduct}=="0a66", MODE:="0666", GROUP:="plugdev"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="8086", ATTRS{idProduct}=="0aa5", MODE:="0666", GROUP:="plugdev"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="8086", ATTRS{idProduct}=="0abf", MODE:="0666", GROUP:="plugdev"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="8086", ATTRS{idProduct}=="0acb", MODE:="0666", GROUP:="plugdev"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="8086", ATTRS{idProduct}=="0ad0", MODE:="0666", GROUP:="plugdev"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="8086", ATTRS{idProduct}=="04b4", MODE:="0666", GROUP:="plugdev"
```

Comment:
On my classic system I see:
```
$ ls -l /dev/video0 
crw-rw----+ 1 root video 81, 0 Feb 23 06:53 /dev/video0

$ udevadm info /dev/video0
P: /devices/pci0000:00/0000:00:14.0/usb1/1-5/1-5:1.0/video4linux/video0
N: video0
S: v4l/by-id/usb-CN09GTFM7248767IB6A8A01_Integrated_Webcam_HD-video-index0
S: v4l/by-path/pci-0000:00:14.0-usb-0:5:1.0-video-index0
E: COLORD_DEVICE=1
E: COLORD_KIND=camera
E: DEVLINKS=/dev/v4l/by-path/pci-0000:00:14.0-usb-0:5:1.0-video-index0 /dev/v4l/by-id/usb-CN09GTFM7248767IB6A8A01_Integrated_Webcam_HD-video-index0
E: DEVNAME=/dev/video0
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-5/1-5:1.0/video4linux/video0
E: ID_BUS=usb
E: ID_FOR_SEAT=video4linux-pci-0000_00_14_0-usb-0_5_1_0
E: ID_MODEL=Integrated_Webcam_HD
E: ID_MODEL_ENC=Integrated_Webcam_HD
E: ID_MODEL_ID=670c
E: ID_PATH=pci-0000:00:14.0-usb-0:5:1.0
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_5_1_0
E: ID_REVISION=5626
E: ID_SERIAL=CN09GTFM7248767IB6A8A01_Integrated_Webcam_HD
E: ID_TYPE=video
E: ID_USB_DRIVER=uvcvideo
E: ID_USB_INTERFACES=:0e0100:0e0200:
E: ID_USB_INTERFACE_NUM=00
E: ID_V4L_CAPABILITIES=:capture:
E: ID_V4L_PRODUCT=Integrated_Webcam_HD
E: ID_V4L_VERSION=2
E: ID_VENDOR=CN09GTFM7248767IB6A8A01
E: ID_VENDOR_ENC=CN09GTFM7248767IB6A8A01
E: ID_VENDOR_ID=0c45
E: MAJOR=81
E: MINOR=0
E: SUBSYSTEM=video4linux
E: TAGS=:uaccess:seat:
E: USEC_INITIALIZED=13275962
```

and if I plugin a usb webcam I get:
```
$ ls -l /dev/video1
crw-rw----+ 1 root video 81, 1 Feb 23 07:11 /dev/video1

$ udevadm info /dev/video1
P: /devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/video4linux/video1
N: video1
S: v4l/by-id/usb-046d_09c1-video-index0
S: v4l/by-path/pci-0000:00:14.0-usb-0:2:1.0-video-index0
E: COLORD_DEVICE=1
E: COLORD_KIND=camera
E: DEVLINKS=/dev/v4l/by-id/usb-046d_09c1-video-index0 /dev/v4l/by-path/pci-0000:00:14.0-usb-0:2:1.0-video-index0
E: DEVNAME=/dev/video1
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/video4linux/video1
E: ID_BUS=usb
E: ID_FOR_SEAT=video4linux-pci-0000_00_14_0-usb-0_2_1_0
E: ID_MODEL=09c1
E: ID_MODEL_ENC=09c1
E: ID_MODEL_ID=09c1
E: ID_PATH=pci-0000:00:14.0-usb-0:2:1.0
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_2_1_0
E: ID_REVISION=0005
E: ID_SERIAL=046d_09c1
E: ID_TYPE=video
E: ID_USB_DRIVER=uvcvideo
E: ID_USB_INTERFACES=:0e0100:0e0200:010100:010200:
E: ID_USB_INTERFACE_NUM=00
E: ID_V4L_CAPABILITIES=:capture:
E: ID_V4L_PRODUCT=UVC Camera (046d:09c1)
E: ID_V4L_VERSION=2
E: ID_VENDOR=046d
E: ID_VENDOR_ENC=046d
E: ID_VENDOR_ID=046d
E: MAJOR=81
E: MINOR=1
E: SUBSYSTEM=video4linux
E: TAGS=:seat:uaccess:
E: USEC_INITIALIZED=1080970387
```

Shouldn't the mode be 0660 and group video? Why did you choose these? What happens if you don't set either? I see in https://github.com/IntelRealSense/librealsense/blob/master/config/99-realsense-libusb.rules that they use the entries you have, but this is not inline with how v4l is typically setup. On an all snaps system without console users, this is less of a concern, but on a classic system, it is. Can you comment?

You might be interested in this bug for reference: https://bugs.launchpad.net/snappy/+bug/1646144